pkg/utils/util.go

```diff
@@ -9,3 +16,66 @@ func Contains(slice []string, value string) bool {
 	}
 	return false
 }
+
+// GetToolchainClusterName produces a name for ToolchainCluster object that is both deterministic and "reasonably unique".
+// The `ordinal` must be greater than 0 if there are multiple ToolchainCluster objects pointing to the same cluster. This
+// needs to be determined by the caller prior to calling this method.
+func GetToolchainClusterName(clusterType, serverAPIEndpoint string, ordinal int) (string, error) {
```

Comment:
This would not behave the same as the script. If the length of the generated name is > 63 and there is no suffix explicitly declared, the add-cluster.sh adds "1" as the suffix. I believe your code would not do that.

Reply:
yup, I missed that detail that we add the `1` suffix to make sure that the name ends with an alphanumeric char :man_facepalming: 

in that case, this should work:
```go
maxAllowedClusterHostNameLen := 62
clusterHostName, err := sanitizeEnpointForUsageAsName(serverAPIEndpoint)
if err != nil {
	return "", fmt.Errorf("failed to sanitize the endpoint for naming purposes: %w", err)
}

clusterNameBase := fmt.Sprintf("%s-%s", clusterType, clusterHostName)
suffix := ""
if ordinal > 0 {
	suffix = fmt.Sprintf("%d", ordinal)
} else if len(clusterNameBase) > maxAllowedClusterHostNameLen {
	suffix = "1"
}

if len(clusterNameBase) > maxAllowedClusterHostNameLen-len(suffix) {
	clusterNameBase = clusterNameBase[0 : maxAllowedClusterHostNameLen-len(suffix)]
}

return fmt.Sprintf("%s%s", clusterNameBase, suffix), nil
```

Reply:
my point is, let's keep the logic simple and just take into account the length of the possible suffix instead of calculating log10 of the ordinal number. 
While this can be a fun and nice mathematical exercise, we still need to make sure that anyone who comes after you to the code can easily understand what is going on there :-)

Reply:
I know it's premature optimization in a wrong place but computing the number of digits using log is approx 50x faster than converting it to string and counting letters, so I kept it in and just added a comment :)

Reply:
Can you see the difference when you execute the command locally :smile: ?